src/aspire/reconstruction/kernel.py

```diff
@@ -160,9 +161,88 @@ def toeplitz(self, L=None):
         if L is None:
             L = int(self.M / 2)
 
-        A = np.eye(L**3, dtype=self.dtype)
+        A = Volume(np.eye(L**3, dtype=self.dtype).reshape((L**3, L, L, L)))
         for i in range(L**3):
-            A[:, i] = np.real(vol_to_vec(self.convolve_volume(vec_to_vol(A[:, i]))))
+            A[i] = self.convolve_volume(A[i])[0]
```

Comment:
Sure, but I'm not sure I understand.  Are you referring to less overhead larger problem vs looping over single volumes?.. or is there another trick?

The fft is already potentially swappable to GPU via `aspire.numeric`.

Reply:
Sorry, was referring to batching the convolutions instead of iterating. Not sure how much we'd save for large volumes, but smaller ones should give a speedup.

Reply:
Looking at `_convolve_volume` this seems straightforward if we can decide how many volumes to stack.  Made issue:

https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/974

Reply:
Thanks!